CONTRIBUTING.rst

```diff
@@ -110,12 +110,39 @@ This section provides general guidance for developing code for the project. The
 - We give support to static type checker for Python >= 3.6
```

Comment:
I might also mention that tests should be grouped by functionalities embed to a class and provide the following:
```python
class TestMyFunction:
    def test_smoke(self):
        # check defaults parameters, i/o shapes
        pass

    def test_feature_foo(self):
        # test basic functionality
        pass

     def test_feature_bar(self):
         # test another functionality
         pass

     def test_gradcheck(self):
         # test the functionality gradients
         pass

     def test_jit(self):
         #  test the functionality using jit modules
         pass